packages/amplify-graphql-schema-generator/src/schema-representation/types.ts

```diff
@@ -0,0 +1,93 @@
+export type FieldType = DefaultType | CustomType | ListType | NonNullType;
+
+export interface DefaultType {
+  readonly kind: 'Scalar';
+  readonly name: 'String' | 'ID' | 'Int' | 'Float' | 'AWSJSON' | 'AWSDate' | 'AWSTime' | 'AWSDateTime' | 'AWSTimestamp';
+}
+
+export interface CustomType {
+  readonly kind: 'Custom';
+  readonly name: string;
+}
+
+export interface ListType {
+  readonly kind: 'List';
+  readonly type: FieldType;
+}
+
+export interface NonNullType {
+  readonly kind: 'NonNull';
+  readonly type: DefaultType | CustomType | ListType;
+}
+
+export interface DefaultValue {
+  readonly kind: 'DB_GENERATED' | 'TRANSFORMER_GENERATED';
+  readonly value: string | number;
+}
+
+export class Field {
+  readonly default: DefaultValue | undefined = undefined;
+  constructor(public name: string, public type: FieldType) {
+  }
+}
+
+export class Index {
+  private fields: string[] = [];
+  constructor(public name: string) {
+  }
+
+  public getFields(): string[] {
+    return this.fields;
+  }
+
+  public setFields(fields: string[]): void {
+    this.fields = [...fields];
+  }
+}
+
+export class Model {
+  private fields: Field[] = [];
+  private primaryKey: Index | undefined = undefined; 
```

Comment:
is undefined explicit assignment needed here?

Reply:
Yes, it is helpful when we use strict typing.